diff --git a/internal/api/settings.go b/internal/api/settings.go
index bc2f3869..c0025f48 100644
--- a/internal/api/settings.go
+++ b/internal/api/settings.go
@@ -36,6 +36,7 @@ type Settings struct {
 	MailerAutoconfirm bool             `json:"mailer_autoconfirm"`
 	PhoneAutoconfirm  bool             `json:"phone_autoconfirm"`
 	SmsProvider       string           `json:"sms_provider"`
+	MFAEnabled        bool             `json:"mfa_enabled"` // preserve for backwards compatibility
 	SAMLEnabled       bool             `json:"saml_enabled"`
 }
 
