diff --git a/internal/api/api.go b/internal/api/api.go
index d9016fd0..924ce018 100644
--- a/internal/api/api.go
+++ b/internal/api/api.go
@@ -175,7 +175,7 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 		})
 
 		r.Route("/sso", func(r *router) {
-			r.Use(api.requireSAMLEnabled)
+			// r.Use(api.requireSAMLEnabled)
 			r.With(api.limitHandler(
 				// Allow requests at the specified rate per 5 minutes.
 				tollbooth.NewLimiter(api.config.RateLimitSso/(60*5), &limiter.ExpirableOptions{
diff --git a/internal/api/saml.go b/internal/api/saml.go
index d936ff2f..011d098c 100644
--- a/internal/api/saml.go
+++ b/internal/api/saml.go
@@ -24,6 +24,7 @@ func (a *API) getSAMLServiceProvider(identityProvider *saml.EntityDescriptor, id
 		externalURL.Path += "/"
 	}
 
+	externalURL.Path += "gotrue/"
 	externalURL.Path += "sso/"
 
 	provider := samlsp.DefaultServiceProvider(samlsp.Options{
